index.d.ts

```diff
@@ -0,0 +1,29 @@
+interface Config {
+    isPromise?: boolean; // is the result a promise
+    maxAge?: number; // amount of time in milliseconds before the cache will expire
+    maxArgs?: number; // maximum number of arguments to use as key for caching
+    maxSize?: number; // maximum size of cache for this method
+    serialize?: boolean; // should the parameters be serialized instead of directly referenced
+    serializeFunctions?: boolean; // should functions be included in the serialization of multiple parameters
```

Comment:
Is the same typing as the return from `createArgumentSerializer`, so I believe it would be `(...args: any[]) => string;`